### PR TITLE
Stats are boxed to support efficient move when canonicalizing

### PR DESCRIPTION
**Pull request by robert3005**

### CODE REVIEW

**Comment by robert3005** [comment]
Locally this shows a modest improvement I will wait for benchmarks

File: vortex-array/src/stats/mod.rs [hunk 1]
```diff
@@ -188,7 +188,9 @@ pub trait Statistics {
 pub trait ArrayStatistics {
     fn statistics(&self) -> &dyn Statistics;
 
-    fn inherit_statistics(&self, parent: &dyn Statistics);
+    fn take_stats(&self) -> Box<StatsSet>;
```
Comment (lwwmanning) [inline vortex-array/src/stats/mod.rs:191]: the fact that this takes `&self` instead of `&mut self` is kind of sketch

Comment (robert3005) [reply vortex-array/src/stats/mod.rs:191]: interior mutability, we don't need `&mut self` since we have `Arc<RwLock<...>>`

**Comment by robert3005** [comment]
This is faster but there's still an issue of into_arrow for a view array. In that case we would really like to skip inheriting stats since that implies a read and deserialization.

File: vortex-array/src/stats/mod.rs (hunk 1 above)
Comment (lwwmanning) [reply vortex-array/src/stats/mod.rs:191]: yeah, I understand how it works rn, but this is a `pub trait`... as API, it feels weird to imply that implementations should have interior mutability?

Comment (robert3005) [reply vortex-array/src/stats/mod.rs:191]: yeah, I'm not sold on this api, just testing things right now. I have found some sketchy things in stats that I'm trying to make better but they require bigger changes

**Comment by robert3005** [comment]
I will come back to this. Inheriting is not as big of an issue as I thought originally and #1652 seems the biggest improvement

**Pull request closed without merging**